crates/bevy_animation/src/lib.rs

```diff
@@ -397,194 +493,222 @@ struct AnimationTargetContext<'a> {
+        for (_, playing_animation) in self.playing_animations_mut() {
+            let new_speed = playing_animation.speed() * factor;
+            playing_animation.set_speed(new_speed);
+        }
         self
     }
 
-    /// Time elapsed playing the animation
-    pub fn elapsed(&self) -> f32 {
-        self.animation.elapsed
+    /// Seeks all active animations forward or backward by the same amount.
+    ///
+    /// To seek forward, pass a positive value; to seek negative, pass a
+    /// negative value. Values below 0.0 or beyond the end of the animation clip
+    /// are clamped appropriately.
+    #[doc(alias = "seek_to")]
+    pub fn seek_all_by(&mut self, amount: f32) -> &mut Self {
+        for (_, playing_animation) in self.playing_animations_mut() {
+            let new_time = playing_animation.seek_time();
+            playing_animation.seek_to(new_time + amount);
+        }
+        self
     }
 
-    /// Seek time inside of the animation. Always within the range [0.0, clip duration].
-    pub fn seek_time(&self) -> f32 {
-        self.animation.seek_time
+    /// Returns the [`ActiveAnimation`] associated with the given animation
+    /// node if it's currently playing.
+    ///
+    /// If the animation isn't currently active, returns `None`.
+    pub fn animation(&self, animation: AnimationNodeIndex) -> Option<&ActiveAnimation> {
+        self.playing_animations.get(&animation)
     }
 
-    /// Seek to a specific time in the animation.
-    pub fn seek_to(&mut self, seek_time: f32) -> &mut Self {
-        self.animation.seek_time = seek_time;
-        self
+    /// Returns a mutable reference to the [`ActiveAnimation`] associated with
+    /// the given animation node if it's currently active.
+    ///
+    /// If the animation isn't currently active, returns `None`.
+    pub fn animation_mut(&mut self, animation: AnimationNodeIndex) -> Option<&mut ActiveAnimation> {
+        self.playing_animations.get_mut(&animation)
     }
 
-    /// Reset the animation to its initial state, as if no time has elapsed.
-    pub fn replay(&mut self) {
-        self.animation.replay();
+    /// Returns true if the animation is currently playing or paused, or false
+    /// if the animation is stopped.
+    pub fn animation_is_playing(&self, animation: AnimationNodeIndex) -> bool {
+        self.playing_animations.contains_key(&animation)
     }
 }
 
 /// A system that advances the time for all playing animations.
 pub fn advance_animations(
     time: Res<Time>,
     animation_clips: Res<Assets<AnimationClip>>,
-    mut players: Query<&mut AnimationPlayer>,
+    animation_graphs: Res<Assets<AnimationGraph>>,
+    mut players: Query<(&mut AnimationPlayer, &Handle<AnimationGraph>)>,
 ) {
-    for mut player in players.iter_mut() {
-        let paused = player.paused;
-        if paused {
+    let delta_seconds = time.delta_seconds();
+    for (mut player, graph_handle) in players.iter_mut() {
```

Comment:
Could `par_iter_mut `be used here to gain performance benefits from parallelization?

Reply:
Done